specs/agents-versioning-publishing.md

```diff
@@ -0,0 +1,62 @@
+# Hypertrace Agents versioning and publishing
+
+This document describes versioning and publishing requirements for Hypertrace agents.
+All Hypertrace agents follow these requirements:
+
+* Semantic versioning for all released artifacts.
+* Release every merge commit into the main or maintenance branch as `"snapshot"`.
+* Maintenance branches for older versions should be named as `release-<major>.<minor>` e.g. `release-1.0`.
+
+#### Semantic versioning
+
+Each released version strictly follows [semantic versioning](https://semver.org/).
+The released version must be tagged in the source code: e.g. `1.0.0` or `v1.0.0` for languages
+with this convention (e.g. Golang).
+
+#### Snapshot release for every merge commit
+
```

Comment:
"(e.g. `release-1.1` for `1.1.x`)" tells you that 1.1.x is a better convention, as you don't need to clarify what it is